src/App.jsx

```diff
@@ -4,13 +4,25 @@ import TodoList from './TodoList';
 import TodoForm from './TodoForm';
 
 function App() {
-  const [newTodo, setNewToDo] = useState('initialValue');
+  const [todoList, setTodoList] = useState([]);
+  const addToDoHandler = (title) => {
+    const newTodo = { title, id: Date.now(), isCompleted: false };
+    setTodoList([...todoList, newTodo]);
+  };
+
+  const onCompleteTodo = (id) => {
+    const updatedTodos = todoList.map((todo) => {
+      return todo.id === id ? { ...todo, isCompleted: true } : todo;
+    });
+
+    setTodoList(updatedTodos);
```

Comment:
This is a nice catch. `updatedTodos` is already a copy of `todoList` so no need to create another copy before passing it to the setter function.